sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFactory.cs

```diff
@@ -27,15 +28,21 @@ namespace SonarAnalyzer.UnitTest.MetadataReferences
 {
     internal static class MetadataReferenceFactory
     {
-        private static readonly string SystemAssembliesFolder = new FileInfo(typeof(object).Assembly.Location).Directory.FullName;
+        private static readonly string SystemAssembliesFolder = Path.GetDirectoryName(typeof(object).Assembly.Location);
 
         public static IEnumerable<MetadataReference> Create(string assemblyName) =>
             ImmutableArray.Create(CreateReference(assemblyName));
 
-        internal static MetadataReference CreateReference(string assemblyName) =>
+        public static IEnumerable<MetadataReference> Create(Type type) =>
+            ImmutableArray.Create(CreateReference(type));
+
+        public static MetadataReference CreateReference(string assemblyName) =>
             MetadataReference.CreateFromFile(Path.Combine(SystemAssembliesFolder, assemblyName));
 
-        internal static MetadataReference CreateReference(string assemblyName, string subFolder) =>
+        public static MetadataReference CreateReference(string assemblyName, string subFolder) =>
```

Comment:
This method can be removed since it's not used.

Reply:
It's used 3 times under `#if NETFRAMEWORK` condition in this file:
https://github.com/SonarSource/sonar-dotnet/blob/master/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/FrameworkMetadataReference.cs#L39